raiden_contracts/contracts/MonitoringService.sol

```diff
@@ -90,18 +90,25 @@ contract MonitoringService is Utils {
 
     /// @notice Deposit tokens used to reward MSs
     /// Can be called by anyone several times and on behalf of other accounts
-    /// @param beneficiary The account benefitting from the deposit
-    /// @param amount The amount of tokens to be depositted
-    function deposit(address beneficiary, uint amount) public {
-        require(amount > 0);
+    /// @param beneficiary The account benefiting from the deposit
+    /// @param total_deposit Idempotent function that sets the total_deposit of
+    //// tokens of the beneficiary
+    function deposit(address beneficiary, uint256 total_deposit) public
+    {
+        require(total_deposit > 0);
+
+        uint256 added_deposit;
+
+        // Calculate the actual amount of tokens that will be transferred
+        added_deposit = total_deposit - balances[beneficiary];
```

Comment:
This is not good. uint256 can only be positive integers, and if the new `total_deposit` is lower than the current balance you have a problem. You must add a check that doesn't allow for this to happen. Ideally you also add a small test to show that it works correctly.

Reply:
Changed.